lib/modules/apostrophe-schemas/index.js

```diff
@@ -977,14 +977,21 @@ module.exports = {
       converters: {
         string: function(req, data, name, object, field, callback) {
           object[name] = self.apos.launder.string(data[name], field.def);
-          if (field.min && (object[name].length < field.min)) {
+          if (object[name] && field.min && (object[name].length < field.min)) {
```

Comment:
It's also possible I have to change `object[name]` to `data[name]` because launder may change the value to the default value and that was not what the user actually input. Which means that it will never fail if the configuration is `min: 2`, `max: 5` and `def: 3` but the user submitted `1` or `null`.

Reply:
There's a whole lot of things to consider here and this is fairly deep inside stuff and not so easy to test manually. I did do many tests to try to cover use cases, but due to my limited insight in Apostrophe I may not be the best person to do this all the way through. So I would for sure appreciate a second set of eyes (or more) and more testing with knowledge of how to provoke this and test it properly. I understand that can delay this PR getting merged and that's okay. I think decoupling this is a good base for doing more converters and validators, for exampel to compare two fields in the same form against each other, conditional so the check is only done if both are shown, with value etc.